app/src/controllers/project/sagas.js

```diff
@@ -430,9 +432,15 @@ function* watchFetchProject() {
 }
 
 function* fetchProjectPreferences({ payload: projectKey }) {
+  const userRoles = yield select(userRolesSelector);
+  const hasFilterPermissions = canWorkWithFilters(userRoles);
+
   const preferences = yield call(fetch, URLS.projectPreferences(projectKey));
   yield put(fetchProjectPreferencesSuccessAction(preferences));
-  yield put(fetchUserFiltersSuccessAction(preferences.filters));
+
+  if (hasFilterPermissions) {
+    yield put(fetchUserFiltersSuccessAction(preferences.filters));
+  }
```

Comment:
In case you are willing to disable the user filter preferences setting to the redux, does it make sense to not send a request for them at all?
I suppose it would be ok to check where `fetchProjectPreferences` called and do not trigger it in case of insufficient permissions.

Reply:
Done.